### PR TITLE
COMMERCE-11732 Test Plan for COMMERCE-11163 Supplier Role to manage their Orders

### PR DESCRIPTION
**Pull request by jaydawu**
Fix conflicts and resent from https://github.com/liferay-commerce/liferay-portal/pull/3967
https://liferay.atlassian.net/browse/COMMERCE-11732

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu Overall tests are good but we need to heavily work on the duration of the tests. Looking at the latest commerce-functional run, there are tests that are going over 10 minutes long and one is around 20 minutes long.
Please investigate how you can improve the test length by changing existing UI configuration with JSON macros.
Another thing I can recommend is to do "Supplier Channel" and "Supplier Catalog" with the order splitting rule only when it's strictly needed by the test(looking at our internal notes, we should have only 2 tests that are using this), for the other tests Minium is enough. Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
The SF issue is not caused by my commits, the fix has been sent [here](https://github.com/brianchandotcom/liferay-portal/commit/99999170bb61854329bd4974c72cd46d6b159be6) (under Bchan's queue).

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase [hunk 1]
```diff
@@ -3249,4 +4409,182 @@ definition {
...
+				key_value = "Order Split by Catalog Action",
+				locator1 = "CommerceEntry#HEALTH_CHECK_ICON_CHECK_NAME");
+
+			CommerceJSONOrdersAPI._patchCommerceOrderObjectAction(
+				active = "true",
+				objectActionName = "SplitOrderByCatalog",
+				objectDefinitionName = "CommerceOrder");
+		}
+
+		task ("When the buyer checkout an order with both Minium and Supplier catalogs products") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase:4511]:
```suggestion
		task ("When the buyer checkout an order with both Test Catalog and Supplier catalogs products") {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu Aside from these 2 renamings I asked, consider it approved.

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
@lorenzolapescara Thank you so much for the review for the past times, I updated the request you asked for the last time, and now the tests are passing, I will forward it.

**Comment by jaydawu** [comment]
ci:forward

**Pull request closed without merging**